### PR TITLE
Handles log roation [DO NOT MERGE]

### LINKED ISSUE #14: handle log rotatation
There are three things that need to happen around database logs, ideally in this order:

1. Query for new events
1. Send logs to CloudWatch Logs
1. Rotate log table

There are a few challenges:

* Ideally(?) these steps would all happen in a single transaction, but since the middle step involves an external API call, that isn't possible.
* Log rotation can be triggered externally in RDS. This means that there may be entries that made it into `general_log` which were then rotated to `general_log_backup` in the time since the script was run last, and thus it will also need to check that table. Per [the documentation](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.Concepts.MySQL.html#Appendix.MySQL.CommonDBATasks.Logs):

    > When `TABLE` logging is enabled, log tables are rotated every 24 hours if the space used by the table logs is more than 20 percent of the allocated storage space or the size of all logs combined is greater than 10 GB. If the amount of space used for a DB instance is greater than 90 percent of the DB instance's allocated storage space, then the thresholds for log rotation are reduced. Log tables are then rotated if the space used by the table logs is more than 10 percent of the allocated storage space or the size of all logs combined is greater than 5 GB.

Working on the best way to solve this.

### PR DESCRIPTION
**Pull request by yfuksenko**
Rotates the logs after they are retrieved from the DB. Send the logs to
cloudwatch in 500 events batches. Closes #14

### CODE REVIEW

File: src/mysql.py [hunk 1]
```diff
@@ -1,58 +1,56 @@
...
-                    FROM general_log
-                    WHERE event_time > %s
-                    ORDER BY event_time DESC
-                    LIMIT %s
-                ) sub
-                ORDER BY event_time ASC
-                """, (since, limit,))
-
-            events = map(mysql_to_cw_log_event, cursor)
-            return list(events)
```
Comment (afeld) [inline src/mysql.py:58]: Not a fan of the `MySQL` class?

**Review by afeld — COMMENTED** [review]
Hmm, a bit hard to follow what changed with all the functions moving around...was that intentional?

File: src/mysql.py (hunk 1 above)
Comment (yfuksenko) [reply src/mysql.py:58]: it looks like something is wrong, don't merge it
